src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -272,14 +275,58 @@ def _prepare_tool(tool_name, work_dir, batch_number, size, node_id, genome_ids,
     return batch_dir, genome_meta
 
 
-def _run_microtrait(genome_id, fna_file, debug):
-    microtrait_result_dict = {'genome_id': genome_id, 'source_file': fna_file}
+def _get_r_list_element(r_list, element_name):
+    # retrieve the element from the R list
+    pos = r_list.names.index(element_name)
+    return r_list[pos]
 
-    # Load the microtrait package
-    # importr("microtrait")
-    # TODO: implement the Rscript to run microtrait, potentially an isolated Rscript and get rid of the rpy2 dependency
 
-    return microtrait_result_dict
+def _unpack_gz_file(gz_file):
+    # unpack the gz file
+    # TODO refactor downloader script to automatically unpack gz files
+
+    output_file_path = os.path.splitext(gz_file)[0]
+    if os.path.exists(output_file_path):
+        print(f'file {output_file_path} already exists. Skipping unpacking {gz_file}')
+        return output_file_path
+
+    print(f'unpacking {gz_file}')
+    with gzip.open(gz_file, 'rb') as f_in:
+        with open(output_file_path, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+    return output_file_path
+
+
+def _run_microtrait(genome_id, fna_file, batch_dir):
+    # run microtrait.extract_traits on the genome file
+
+    genome_dir = os.path.join(batch_dir, genome_id)
+    os.makedirs(genome_dir, exist_ok=True)
+
+    # Load the R script as an R function
+    r_script = """
+    library(microtrait)
+    extract_traits <- function(genome_file, out_dir) {
+      genome_file <- file.path(genome_file)
+      microtrait_result <- extract.traits(in_file = genome_file, out_dir = out_dir)
+      return(microtrait_result)
+    }
+    """
+    r_func = robjects.r(r_script)
+
+    if fna_file.endswith('.gz'):
+        fna_file = _unpack_gz_file(fna_file)
```

Comment:
As above, this should be deleted when done

Reply:
👍

Reply:
:+1: